### DISCUSSION TITLE
RTL_433 ASK Manchester?

### DISCUSSION THREAD

**Discussion by Sandman2001** [merbanan/rtl_433#3001] [Q&A]
I am currently working with an Arduino R3 and an [Amplitude Shift Keying(ASK) 4800 bps receiver](https://www.sparkfun.com/products/10532). For the device I'm working to collect data from, it is a pressure sensor that uses ASK Manchester (IEEE 802.3) Encoding at 434MHz w/ 4000 bps 125 uS nibble time (ex. 125uS low, then 125uS high for a single bit). I have done some digging and found an "ASK" implementation and "FSK Manchester" in the library, but nothing specific to "ASK Manchester" encoding. For those experienced with the library, does the encoding need to be accounted for in software? At this stage of implementation, I am looking to receive a raw bit stream from the sensor to confirm setup.

I started out with the Arduino extension of the RTL_433 library, specifically the RH_ASK library. I have a tool for the sensor that gets the device ID and other information, so i know that the sensor side is working. I have an oscilloscope connected to the digital out of the rf receiver, i have confirmed that the signal is being picked up by the Rx, a concern I have with it is that it may not be hitting proper logic levels (scope shots below). I am also unaware of how the recv and available functions work for the RF library. I have confirmed that the digital pin on the arduino is seeing the logic transistions.

![A single burst of data](https://github.com/user-attachments/assets/92e27a63-9584-4f6f-9057-af364b8ea6ee)
^A single burst of data
![zoomed on the logic shifts](https://github.com/user-attachments/assets/f9869d2b-4f9b-4091-9f9b-04f40a65e038)
^zoomed on the logic shifts
![timing errors? look at spacing between shifts](https://github.com/user-attachments/assets/5bef9ee1-aba1-4e72-9747-ca172393f2c9)
^timing errors? look at spacing between shifts

Update: questioning implementing this [library](https://mchr3k.github.io/arduino-libs-manchester/)

![wave processed through ASK Recv, seen through digital out pin](https://github.com/user-attachments/assets/fd835e42-1a10-4b8d-9434-0a0123a150aa)
^wave processed through ASK Recv, seen through digital out pin

![sets of information being sent, can see data within each pulse](https://github.com/user-attachments/assets/003d02e3-104d-4b0d-b43e-efeda3def6d1)
^sets of information being sent, can see data within each pulse

![a single pulse with data enclosed](https://github.com/user-attachments/assets/c5d761cd-384c-4c43-9502-88a07d005072)
^a single pulse with data enclosed

![a single bit from transmission, 250 uS time for one bit, split into 125 uS on/off sections for Manchester encoding](https://github.com/user-attachments/assets/37b59ca3-c255-4b65-b74b-8050b09e24a4)
^a single bit from transmission, 250 uS time for one bit, split into 125 uS on/off sections for Manchester encoding

![another data packet from same transmission, this one has iffy logic levels](https://github.com/user-attachments/assets/daa3fa23-3dd5-49b9-bd95-2e5a943b5277)
^another data packet from same transmission, this one has iffy logic levels

![scope shot of potentially unreadable bits or noise](https://github.com/user-attachments/assets/57190fdd-d14d-4b09-8b95-ec6e094c2adc)
^scope shot of potentially unreadable bits or noise

**Comment by zuckschwerdt** [comment]
OOK and MC are different layers. You can mix and match.
See https://triq.org/rtl_433/PULSE_FORMATS.html#mc-%E2%80%94-manchester-code for some codings (mouse wheel to zoom).

The scope shots look like noise.

"ASK receiver" might be exaggerated ;) It looks like those cheap Superheterodyne OOK sets (technically OOK is a kind of 2-ASK)
Maybe an I2C transceiver chip like the CC1101 or RFM69 or similar would be better suited?

**Reply by Sandman2001** [reply]
When I'm looking at the scope shots on continuous mode there is a bunch of noise (as seen surrounding the more defined signal shot), but when I ping the TPMS sensor it seems to clear up. I don't know much about rf and how this receiver works, but I would not be surprised if the receiver I am using is not sufficient to process the signal. Being a RF newbie, a question I had is if ASK is interchangeable with OOK? Also, I have a few other rf receivers at home, will link them in another comment. Would appreciate some assistance picking out one best suited for this application.

**Reply by zuckschwerdt** [reply]
https://en.wikipedia.org/wiki/Amplitude-shift_keying is the general term for things like OOK. With binary ASK or 2-ASK you have two amplitude levels, OOK simply has those levels as 0% and 100%.

Those cheap receivers will see data in the noise when the band is silent. It might work if you know how to look for valid data in a continuous stream of noise and sometimes signal. It won't help to trigger on the data pin, you would need to read all transitions and try to find a valid packet of data.

**Reply by Sandman2001** [reply]
I did some digging and found I have an [adafruit 32u4 rh68chw](https://learn.adafruit.com/adafruit-feather-32u4-radio-with-rfm69hcw-module/overview). It seems like this will work for both frequencies of my projects listed in the discussion (433/915MHz). A few questions for you: have you had any experience with this specific MC/ radio module combo? There is always hardware specific initialization you have to do for each MC, but is the software relatively plug and play with little fuss? if not, are there any tips and/or tricks you learned to expedite set up of this library with HW you are new to?

**Reply by zuckschwerdt** [reply]
There are good tutorials and libs for the RFM69. I have not used that chip yet.

If you mean to port rtl_433 to the ATmega32u4 and RFM69 that would be quite some coding.
But you can push the demodulated data to a computer running rtl_433.

For just one known protocol I'd rather decode the received data inplace (skipping rtl_433) and be done.

**Reply by Sandman2001** [reply]
I see someone has made an attempt to [port over rtl_433 to arduino/lilygoo ](https://github.com/NorthernMan54/rtl_433_ESP). I also have a [lilygoo ax2101 433MHz](https://usa.banggood.com/LILYGO-Meshtastic-AXP2101-T-Beam-V1_2-ESP32-LoRa-Development-Board-433MHz-868MHz-915MHz-923MHz-WiFi-Bluetooth-GPS-OLED-Display-p-1992938.html?gmcCountry=US&currency=USD&cur_warehouse=CN&createTmp=1&ID=513922&utm_source=bing_pa&utm_medium=cpc_us&utm_content=sandra&utm_campaign=aceng-bpla-usg-all-230707&ad_id=&msclkid=c9403564a2c7198f7cd153fa341b15f4). I will be giving this combination a try. Before I fall too far down the rabbit hole, I noticed the radio module was a SX1278, which is not one of the listed recommendations from you. Are there hardware limitations to these receivers that I should be aware of and choose based on modulation type of the carrier wave?

**Reply by zuckschwerdt** [reply]
Note that none of those transceivers readily work with rtl_433. These modules already do half of rtl_433's jobs, detecting and demodulating signals. rtl_433 can then be used to pick the data apart, but really you can do that yourself easily. E.g. look at the description here https://github.com/merbanan/rtl_433/blob/master/src/devices/bresser_3ch.c and pull then bits apart like done there.
